### PR TITLE
multipath after registration for 15-SP2+

### PR DESCRIPTION
**Pull request by klwagner**
Conditionally change order: multipath activation dialog to
come up only after the SCC registration dialog to maintain
compatibility with the SLE-15 SP2 installer.

- Related ticket: https://bugzilla.suse.com/show_bug.cgi?id=1165159
- Needles: (none)
- Verification run (SLE-15 SP2): [client (1053)](http://polya.suse.de/tests/1053), [supportserver (1052)](http://polya.suse.de/tests/1052)

_NOTE_  The following unsuccessful job pair demonstrates that the attempt to defined the modified scheduling for the mru-install-multipath-remote job by means of a *.yaml schedule file (job variable `YAML_SCHEDULE`) is improper (as of now). _Reason:_ the *.yaml schedule would be  forwarded to the parallel supportserver job, too, which is completely unwanted and bogus.
[client 1051](http://polya.suse.de/tests/1051), [supportserver 1050](http://polya.suse.de/tests/1050)

Therefore going the traditional way of a conditional reschedule via `main_common.pm` seems inevitable.

### CODE REVIEW

**Comment by klwagner** [comment]
Q to reviewer: It might be that the conditional: `is_sle('15-SP2+')` is too restrictive. However, I'm just not aware of any other new products which would feature the same modified order. Comments welcome.

**Comment by okurz** [comment]
What about a corresponding TW test? Because of [Factory First](https://opensource.suse.com/suse-open-source-policy) we should test against Tumbleweed by default (with exceptions for special packages, of course). Could you please check if it works with a current TW snapshot and then add to the main.pm there? This should answer your question for how a "newer product" behaves :)

**Comment by klwagner** [comment]
Thank you for the hint regarding method and policy.

Now it turns out that the latest Tumbleweed media [DVD - 20200307 iso](ftp://dist.suse.de/install/openSUSE-Tumbleweed/openSUSE-Tumbleweed-DVD-x86_64-Snapshot20200307-Media.iso) and [NET - 20200307 iso](ftp://dist.suse.de/install/openSUSE-Tumbleweed/openSUSE-Tumbleweed-NET-x86_64-Snapshot20200307-Media.iso) announces itself as the old SLE-15 SP2 Snapshot 6 (Build 143.1).

- I've already got successful earlier openQA jobs for [Build 143.1](http://polya.suse.de/tests/overview?distri=sle&version=15-SP2&build=143.1&groupid=12): [client 1039](http://polya.suse.de/tests/1039), [supportserver 1038](http://polya.suse.de/tests/1038)
- by now, these openQA jobs for 143.1 -- or Tumbleweed -- seem to not work well anymore since the needed [registration server](http://all-143.1.proxy.scc.suse.de) would not provide valid repositories for this old build anymore.
- On the other hand, a manual installation from the Tumbleweed DVD medium -- with openQA-like VMs as described in [bsc#1165159](https://bugzilla.suse.com/show_bug.cgi?id=1165159#c1) confirmed that Tumbleweed as well features the order of the multipath step now coming up _after_ the scc registration.
- finally, openSUSE Leap does not feature a scc registration dialog at all, and the multipath dialog immediately follows the disk activation (iSCSI in this case).

Am I correct in assuming that `is_sle('15-SP2+')` is also true for Tumbleweed? `lib/version_utils.pm` appears to indicate so.

If so, it appears that all that is left to do in order to also cover Tumbleweed is to move the second `loadtest "installation/multipath";` out of the `if (is_sle)` conditional:

 `if (is_sle) {`
  
`        loadtest 'installation/network_configuration' if get_var('NETWORK_CONFIGURATION');`

`          loadtest "installation/scc_registration";`

`}`

`if (is_sle('15-SP2+') and (get_var('MULTIPATH') or get_var('MULTIPATH_CONFIRM'))) {`

`    loadtest "installation/multipath";`

`}`

`if (is_sle) {`

`        if (is_sles4sap and is_sle('<15') and !is_upgrade()) { `
` ...`

If you agree, I could prepare a corresponding amendment for this PR.

**Comment by okurz** [comment]
> Now it turns out that the latest Tumbleweed media DVD - 20200307 iso and NET - 20200307 iso announces itself as the old SLE-15 SP2 Snapshot 6 (Build 143.1).

Not sure how you come to that conclusion. Maybe it helps to crosscheck with your fellow team members within your team on how they approach something like this? 

> Am I correct in assuming that is_sle('15-SP2+') is also true for Tumbleweed? lib/version_utils.pm appears to indicate so.

No, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/version_utils.pm#L281 that clearly returns "0", i.e. a false value, for any "DISTRI" value that is not "sle". For any current Tumbleweed test, e.g. see https://openqa.opensuse.org/tests/overview?distri=opensuse&groupid=1&version=Tumbleweed , the value of "DISTRI" should be "opensuse".
Please keep in mind that our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. So at best avoid any `is_tumbleweed`. Keeping the `is_sle` for anything that differs in SLE-specific behaviour is fine.

**Comment by klwagner** [comment]
I had to migrate the test in question to Tumbleweed (needs an add-on to `opensuse/main.pm` and a couple of needles). Amended PR to come up RSN.

Tumbleweed verification runs: [client](http://polya.suse.de/tests/1076), [supportserver](http://polya.suse.de/tests/1075)

**Comment by klwagner** [comment]
> > Now it turns out that the latest Tumbleweed media DVD - 20200307 iso and NET - 20200307 iso announces itself as the old SLE-15 SP2 Snapshot 6 (Build 143.1).
> 
> Not sure how you come to that conclusion.

This turned out to be a temporary condition for this build only. The next Tumbleweed media were not SLE-like anymore. No need to dwell on it.

> > Am I correct in assuming that is_sle('15-SP2+') is also true for Tumbleweed? lib/version_utils.pm appears to indicate so.
> 
> No, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/version_utils.pm#L281 that clearly returns "0", i.e. a false value, for any "DISTRI" value that is not "sle".

OK, point taken. This means that the test would have failed for that one SLE-like Tumbleweed build: but it went on to succeed later for the next build, like in the provided verification runs.

**Comment by klwagner** [comment]
Associated: [openSUSE needles PR#652](https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/652)

File: lib/main_common.pm [hunk 1]
```diff
@@ -858,7 +858,8 @@ sub load_inst_tests {
     if (get_var('ENCRYPT_CANCEL_EXISTING') || get_var('ENCRYPT_ACTIVATE_EXISTING')) {
         loadtest "installation/encrypted_volume_activation";
     }
-    if (get_var('MULTIPATH') or get_var('MULTIPATH_CONFIRM')) {
+    # SLE-15 SP2 pushes the multipath activation dialog after the scc_registration dialog, see below
+    if ((get_var('MULTIPATH') or get_var('MULTIPATH_CONFIRM')) and not is_sle('15-SP2+')) {
```
Comment (rwx788) [inline lib/main_common.pm]: Instead of negation here you can simply write `('<15-SP2')`

File: products/opensuse/main.pm [hunk 2]
```diff
@@ -347,6 +347,9 @@ elsif (get_var("SUPPORT_SERVER")) {
     loadtest "support_server/boot";
     loadtest "support_server/login";
     loadtest "support_server/setup";
+    loadtest "support_server/meddle_multipaths" if (get_var("SUPPORT_SERVER_TEST_INSTDISK_MULTIPATH"));
```
Comment (rwx788) [inline products/opensuse/main.pm]: As this looks unrelated to the motivation of the change mentioned of PR, could you please move it to the separate commit?

File: lib/main_common.pm (hunk 1 above)
Comment (klwagner) [reply lib/main_common.pm]: The idea of this formulation is to forestall the concern that the `loadtest "installation/multipath"` invocation might have been made more restricted than before. The conditions `not is_sle('15-SP2+')` and later `if (is_sle('15-SP2+') ...` are clearly complementary, so this is obviously not the case. Your proposal `('<15-SP2')` is equivalent, but just slightly less obvious IMHO -- a matter of taste. What do you think?

File: products/opensuse/main.pm (hunk 2 above)
Comment (klwagner) [reply products/opensuse/main.pm]: Fair enough. I'll comply in a PR amendment. This means, though, that the Tumbleweed verification run demanded by another reviewer is no longer possible here, but only after both commits.

File: lib/main_common.pm (hunk 1 above)
Comment (rwx788) [reply lib/main_common.pm]: It's indeed matter of taste, so we can leave it as it is. Just wanted to be sure that you are aware of that </> syntax.

File: products/opensuse/main.pm (hunk 2 above)
Comment (klwagner) [reply products/opensuse/main.pm]: As required, submitted for opensuse: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9783
